sinks/sinks_service.go

```diff
@@ -104,19 +90,35 @@ func (svc sinkService) ViewSink(ctx context.Context, token string, key string) (
 }
 
 func (svc sinkService) ListSinks(ctx context.Context, token string, pm PageMetadata) (Page, error) {
-	res, err := svc.auth.Identify(ctx, &mainflux.Token{Value: token})
+	res, err := svc.identify(token)
 	if err != nil {
 		return Page{}, errors.Wrap(errors.ErrUnauthorizedAccess, err)
 	}
 
-	return svc.sinkRepo.RetrieveAll(ctx, res.GetId(), pm)
+	return svc.sinkRepo.RetrieveAll(ctx, res, pm)
 }
 
 func (svc sinkService) DeleteSink(ctx context.Context, token string, id string) error {
-	res, err := svc.auth.Identify(ctx, &mainflux.Token{Value: token})
+	res, err := svc.identify(token)
 	if err != nil {
 		return errors.Wrap(errors.ErrUnauthorizedAccess, err)
 	}
 
-	return svc.sinkRepo.Remove(ctx, res.GetId(), id)
+	return svc.sinkRepo.Remove(ctx, res, id)
+}
+
+func validateBackend(sink Sink) error {
```

Comment:
this parameter has to be passed by pointer since the goal is to modify the object. this was caught by LGTM static analysis: https://lgtm.com/projects/g/ns1labs/orb/rev/pr-70e8d1c4f6ce1cf6c73de8a1743874fd88ecaa22